iceoryx_hoofs/include/iceoryx_hoofs/internal/cxx/optional.inl

```diff
@@ -107,12 +112,16 @@ inline optional<T>& optional<T>::operator=(const optional& rhs) noexcept
         {
             construct_value(rhs.value());
         }
+        else
+        {
+            // do nothing
+        }
```

Comment:
@saif-at-github this does not look nice. Which rule does cause the addition of this branch?

Reply:
I agree, this does not improve readability or decreases complexity.

Reply:
the rule is `AutosarC++19_03-M6-4-2` - "All if ... else if constructs shall be terminated with an else clause."